### PR TITLE
Mobile: Reenable and refactor deep linking

### PR DESCRIPTION
**Pull request by cvarley100**
# Description

- Moves mobile deep linking to HOC

To do:
- [ ] Move desktop deep linking to HOC
- [x] Add migration function for `deepLinking` setting
- [x] Add mobile UI
- [ ] Add desktop UI

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?

- Tested on Android and iOS

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes
- [ ] For changes to `shared`: If applicable, I have verified that my changes are implemented correctly in `desktop` and `mobile`

### CODE REVIEW

File: src/shared/schema/index.js [hunk 1]
```diff
@@ -286,6 +286,13 @@ export const WalletSettingsSchema = {
             type: 'bool',
             default: false,
         },
+        /**
+         * Determines if deep linking is enabled
+         */
+        deepLinking: {
```
Comment (rajivshah3) [inline src/shared/schema/index.js]: @laumair do we need to bump the `schemaVersion` for this change?

Comment (cvarley100) [reply src/shared/schema/index.js]: Yeah, and we will need to write a migration. @laumair is gonna refactor migrations as a whole, then we can add this one.

**Comment by laumair** [comment]
@cvarley100 Is this only awaiting migrations?

**Comment by cvarley100** [comment]
@laumair No, it also needs settings UI and associated checks.

File: src/mobile/src/ui/views/wallet/DeepLinking.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        completeDeepLinkRequest: PropTypes.func.isRequired,
+    };
+
+    constructor() {
+        super();
+        this.onChange = this.onChange.bind(this);
+    }
+
+    componentDidMount() {
+        leaveNavigationBreadcrumb('Pow');
```
Comment (rajivshah3) [inline src/mobile/src/ui/views/wallet/DeepLinking.js]:
```suggestion
        leaveNavigationBreadcrumb('DeepLinking');
```

File: src/mobile/src/ui/components/DeepLinking.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import { changeHomeScreenRoute } from 'shared-modules/actions/home';
+
+export default () => (C) => {
+    class WithDeepLinking extends Component {
+        constructor() {
+            super();
+            this.setDeepUrl = this.setDeepUrl.bind(this);
+        }
+        componentWillMount() {
+            this.deepLinkSub = Linking.addEventListener('url', this.setDeepUrl);
```
Comment (rajivshah3) [inline src/mobile/src/ui/components/DeepLinking.js]: Does this listener need to be removed on unmount?

**Review by rajivshah3 — CHANGES_REQUESTED** [review]

File: src/mobile/src/ui/components/DeepLinking.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import { generateAlert } from 'shared-modules/actions/alerts';
+import { changeHomeScreenRoute } from 'shared-modules/actions/home';
+
+export default () => (C) => {
+    class WithDeepLinking extends Component {
+        constructor() {
+            super();
+            this.setDeepUrl = this.setDeepUrl.bind(this);
+        }
+        componentWillMount() {
```
Comment (rajivshah3) [inline src/mobile/src/ui/components/DeepLinking.js]: Tiny nitpick:
```suggestion
        componentDidMount() {
```
`componentWillMount` will be deprecated in a future release of React, so the [docs](https://reactjs.org/docs/react-component.html#unsafe_componentwillmount) advise using `componentDidMount` for subscriptions

**Review by rajivshah3 — APPROVED** [review]

**Pull request merged**